src/js/controllers/imageCtrl.js

```diff
@@ -0,0 +1,35 @@
+(function (angular) {
+    angular
+        .module('flickrPOC')
+        .controller('imagesCtrl', imagesCtrl);
+
+    imagesCtrl.$inject = ['images'];
+
+    function imagesCtrl(images) {
+        /* jshint validthis: true */
+        var vm = this;
+
+        vm.getImages = function () {
+            images.loadPics()
+                .then(function (res) {
+                    //console.log('this is not from test '+ res);
+                    vm.images = res;
+                    return vm.images;
+                })
+                .catch(
+                    function(err){
+                        console.error('Error loading images ', err.status, err.data);
+                    })
+                .finally(function () {
+                    console.log("finally finished");
+                });
+
+        };
+        function activate(){
+            vm.getImages();
```

Comment:
Are you planning on using `getImages()` in the HTML on a button click or similar? If so, then no worries. But, if not, and `vm.getImages()` only exists to be called internally by the controller, then we can remove the method from the `vm` and simply make the `images.loadPics()` call in the `activate() `function.

```javascript
function activate(){
  images.loadPics()
    .then()
    .catch()
}
```

This change will require some updates to the way the tests are run. See my comments on that file for details.